spec/mysql2/client_spec.rb

```diff
@@ -6,13 +6,15 @@
     let(:cnf_file) { File.expand_path('../../my.cnf', __FILE__) }
 
     it "should not raise an exception for valid defaults group" do
+      @client.close
       expect {
         opts = DatabaseCredentials['root'].merge(:default_file => cnf_file, :default_group => "test")
         @client = Mysql2::Client.new(opts)
       }.not_to raise_error
     end
 
     it "should not raise an exception without default group" do
+      @client.close
```

Comment:
Perhaps clearer to have these tests simply not set `@client`. Let them be responsible for creating and closing this separate client instance.